scripts/migrate-flattened.js

```diff
@@ -0,0 +1,82 @@
+/* eslint-disable camelcase */
+/* eslint-disable new-cap */
+/* eslint-disable no-multi-str */
+/* eslint-disable no-template-curly-in-string */
+
+require("dotenv").config()
+
+const exec = require("child_process").exec
+const fs = require("fs")
+const os = require("os")
+const cli = new cli_func()
+
+if (process.argv.length < 4) {
+  console.log()
+  console.log("\n\
+    Usage: yarn migrate-flattened <ArtifactName> <Network>\n\
+       or: npm run migrate-flattened <ArtifactName> <Network>\n\n\
+  ")
+  process.exit(0)
+}
+
+const artifact = process.argv[2]
+const network = process.argv[3]
+process.env.FLATTENED_DIRECTORY = `./flattened/${artifact}/`
+
+if (!fs.existsSync(`${process.env.FLATTENED_DIRECTORY}/Flattened${artifact}.sol`)) {
+  console.log("\n\
+    > Please, flatten the artifact first. E.g.:\n\
+      $ yarn flatten contracts" + (os.type() === "Windows_NT" ? "\\" : "/") + artifact + ".sol\n\n\
+  ")
+  process.exit(0)
+}
+
+compileFlattened().then(() => {
+  console.log()
+  migrateFlattened(network).then(() => {
+    console.log()
+  })
+})
+  .catch(err => {
+    console.error("Fatal:", err)
+    console.error()
+    process.exit(-1)
+  })
+
+/// ////////////////////////////////////////////////////////////////////////////
+
+function cli_func () {
+  this.exec = async function (cmd) {
+    return new Promise((resolve, reject) => {
+      exec(cmd, (error, stdout, stderr) => {
+        if (error) {
+          reject(error)
+          return
+        }
+        resolve(stdout)
```

Comment:
Just a styling suggestion. Early returns is only justified when you need to prevent nesting in several levels. Otherwise, conditionals are more readable because they retain the "single exit point" principle.
```suggestion
        if (error) {
          reject(error)
        } else {
          resolve(stdout)
         }
```